pkg/sanitizer/helpers.go

```diff
@@ -8,38 +8,61 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
-const blocked = "blocked"
+// ClusterInfo holds the provider, capacity, and capabilities.
+type ClusterInfo struct {
+	Provider     string
+	Capacity     int
+	Capabilities []string
+}
 
-type ClusterMap map[string]string
+// ClusterMap maps a cluster name to its corresponding ClusterInfo.
+type ClusterMap map[string]ClusterInfo
 
-func LoadClusterConfig(filePath string) (ClusterMap, sets.Set[string], error) {
-	data, err := os.ReadFile(filePath)
-	if err != nil {
-		return nil, nil, err
+func loadClusterConfigFromBytes(data []byte) (ClusterMap, sets.Set[string], error) {
+	var clusters map[string][]struct {
+		Name         string   `yaml:"name"`
+		Capacity     int      `yaml:"capacity"`
+		Capabilities []string `yaml:"capabilities"`
+		Blocked      bool     `yaml:"blocked"`
 	}
-
-	var clusters map[string][]string
-	err = yaml.Unmarshal(data, &clusters)
-	if err != nil {
+	if err := yaml.Unmarshal(data, &clusters); err != nil {
 		return nil, nil, err
 	}
-
 	blockedClusters := sets.New[string]()
 	clusterMap := make(ClusterMap)
+
 	for provider, clusterList := range clusters {
-		if provider != blocked {
-			for _, cluster := range clusterList {
-				clusterMap[cluster] = provider
+		for _, cluster := range clusterList {
+			if cluster.Capacity == 0 || cluster.Capacity > 100 {
+				cluster.Capacity = 100
+			} else if cluster.Capacity < 0 {
+				cluster.Blocked = true
+			}
+			if cluster.Blocked {
+				blockedClusters.Insert(cluster.Name)
+				continue
+			}
+			clusterMap[cluster.Name] = ClusterInfo{
+				Provider:     provider,
+				Capacity:     cluster.Capacity,
+				Capabilities: cluster.Capabilities,
 			}
-		}
-		if provider == blocked {
-			blockedClusters.Insert(clusterList...)
 		}
 	}
 
 	return clusterMap, blockedClusters, nil
 }
 
+// LoadClusterConfig loads cluster configuration from a YAML file, returning a ClusterMap and a set of blocked clusters.
+func LoadClusterConfig(filePath string) (ClusterMap, sets.Set[string], error) {
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, nil, err
+	}
+	return loadClusterConfigFromBytes(data)
+
```

Comment:
Remove this blank line.

Reply:
If that will be the only discussed change, I will do it in the follow-up PR.